File: prime.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
#include <fstream>

bool* SieveOfEratosthenes(unsigned long n)
{
    bool* A = new bool[n+1];
    // std::fill(std::begin(A), std::end(A), true);
    for (unsigned long i = 0; i < n; i++)
    {
        A[i] = true;
    }
    
    for (unsigned long i = 2; i < sqrt(n); i++)
    {
        if (A[i] == true)
        {
            unsigned long j = pow(i, 2);
            while (j <= n)
            {
                A[j] = false;
                j+=i;
            }
        }
    }
    return A;
}

int main()
{
    unsigned long N = pow(10, 6);
    bool* A = SieveOfEratosthenes(N);
    std::ofstream outputFile;
    outputFile.open("primecpp.txt");
    for (unsigned long i = 0; i < N; i++)
    {
        outputFile << i << ": " << A[i] << "\n";
    }
    outputFile.close();
    return 0;
}

File: prime.py
import math

def SieveOfEratosthenes (n):
    A = []
    A = [True for i in range(n+1)]
    for i in range(2, int(math.sqrt(n))):
        if A[i] == True:
            j = i**2
            while j <= n:
                A[j] = False
                j += i
    return A

N = 10**6
A = SieveOfEratosthenes(N)
outputFile = open("primepy.txt", 'w')
for i in range(N):
    outputFile.write(str(i) + ": " + str(A[i]) + "\n")
outputFile.close()
        